lib/charms/lego_base_k8s/v0/lego_client.py

```diff
@@ -130,44 +123,96 @@ def __init__(self, *args, plugin: str):
             self.tls_certificates.on.certificate_creation_request,
             self._on_certificate_creation_request,
         )
+        self.framework.observe(self.on.config_changed, self._sync_certificates)
         self.framework.observe(self.on.update_status, self._sync_certificates)
+        self.framework.observe(self.on.collect_unit_status, self._on_collect_status)
         self._plugin = plugin
 
-    def validate_generic_acme_config(self) -> bool:
-        """Validate generic ACME config."""
-        if not self._email:
-            self.unit.status = BlockedStatus("Email address was not provided")
-            return False
-        if not self._server:
-            self.unit.status = BlockedStatus("ACME server was not provided")
-            return False
-        if not self._email_is_valid(self._email):
-            self.unit.status = BlockedStatus("Invalid email address")
-            return False
-        if not self._server_is_valid(self._server):
-            self.unit.status = BlockedStatus("Invalid ACME server")
-            return False
-        return True
+    def _on_collect_status(self, event: CollectStatusEvent) -> None:
+        """Handle the collect status event."""
+        if not self._container.can_connect():
+            event.add_status(WaitingStatus("Waiting to be able to connect to vault unit"))
+            return
+
+        if (err := self.validate_generic_acme_config()):
+            event.add_status(
+                BlockedStatus(err)
+            )
+            return
+        if (err := self._validate_plugin_config()):
+            event.add_status(
+                BlockedStatus(err)
+            )
+            return
+        event.add_status(ActiveStatus())
+
+    def _sync_certificates(self, event: EventBase) -> None:
+        """Goes through all the certificates relations and handles outstanding requests."""
```

Comment:
```suggestion
        """Go through all the certificates relations and handle outstanding requests."""
```

Reply:
Does the linter ignore this file? I believe it should catch this category of lints

Reply:
Seems like the choice of word here is unfortunate ("Goes"), the lint checker does indeed catch these errors, it didn't for this word. (tested a different word :D )
However I found other issues we had with the config of ruff and pyright, I fixed them, please take a look :)